docs/reference/components/sinks/amqp.cue

```diff
@@ -0,0 +1,106 @@
+package metadata
+
+components: sinks: amqp: {
+	title: "Amqp"
+
+	classes: {
+		commonly_used: true
+		delivery:      "at_least_once"
+		development:   "stable"
+		egress_method: "dynamic"
+		service_providers: ["AWS", "Confluent"]
+		stateful: false
+	}
+
+	features: {
+		buffer: enabled:      true
+		healthcheck: enabled: true
+		send: {
+			batch: {
+				enabled:      true
+				common:       true
+				max_bytes:    null
+				max_events:   null
+				timeout_secs: null
+			}
+			compression: {
+				enabled: true
+				default: "none"
+				algorithms: ["none", "gzip", "lz4", "snappy", "zstd"]
+				levels: ["none", "fast", "default", "best", 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+			}
+			encoding: {
+				enabled: true
+				codec: {
+					enabled: true
+					default: null
+					enum: ["json", "text"]
+				}
+			}
+			request: enabled: false
+			tls: {
+				enabled:                true
+				can_enable:             true
+				can_verify_certificate: false
+				can_verify_hostname:    false
+				enabled_default:        false
+			}
+			to: components._amqp.features.send.to
+		}
+	}
+
+	support: components._amqp.support
+
+	configuration: {
+	    connection: {
+	        common: true
+	        description: "Connection options for Amqp sink"
+	        required: true
+	        warnings: []
+	        type: object: {
+	            user: components._amqp.configuration.user
+	            password: components._amqp.configuration.user
+	            host: components._amqp.configuration.user
+	            port: components._amqp.configuration.user
+	            connection_timeout: components._amqp.configuration.user
+	        }
+	    }
+	    exchange: {
+            description: "The exchange to publish messages to"
+            required:    true
+            warnings: []
+            type: string: {
+                examples: ["message_exchange"]
+                syntax: "literal"
+            }
+        }
+		key_field: {
```

Comment:
What would you think of making this a templatable field and calling it `queue_key`? This would let users use the template syntax like `queue_key = "{{ some_field }}-{{some other field}}"`?

I am noticing that the docs for the `kafka` sink `topic` field is not quite right, but it is templatable and could be used as an example for implementation here: https://github.com/timberio/vector/blob/master/src/sinks/kafka.rs#L43

Reply:
Still wondering about this too.

Reply:
Let me know if I did the templating right.